File: Assignment6.cpp
#include<bits/stdc++.h>
using namespace std;

#define endl "\n"

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(0); cout.tie(0);
	int n, p, f;
	int count = 0;
	cin >> n >> p >> f;
	// int morning[n], evening[n];
	vector<int> morning(n), evening(n);
	for(int i = 0; i < n; i++) {
		cin >> morning[i];
	}
	for(int i = 0; i < n; i++) {
		cin >> evening[i];
	}
	sort(morning.begin(), morning.end());
	sort(evening.rbegin(), evening.rend());
	for(int i = 0; i < n; i++) {
		if(morning[i] + evening[i] > p) {
			count += morning[i] + evening[i] - p;
		}
	}
	cout << count * f;
	return 0;
}

File: Assignment6.py
def solve():
    n, p, f = input().split(" ")
    n = int(n)
    p = int(p)
    f = int(f)
    morning = list(input().split(" "))
    evening = list(input().split(" "))
    #  print(morning, evening)
    #  morning = [int(x) for x in morning]
    for i in range(len(morning)):
        morning[i] = int(morning[i])
        evening[i] = int(evening[i])
    #  evening = [int(x) for x in evening]
    morning.sort(); evening.sort()
    #  print(morning)
    #  print(evening)
    count = 0
    for i in range(len(morning)):
        if morning[i] + evening[len(evening) - i - 1] > p:
            count += morning[i] + evening[len(evening) - i - 1] - p
    print(count * f)

solve()
